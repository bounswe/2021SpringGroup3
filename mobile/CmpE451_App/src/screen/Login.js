import React, {useState, useEffect, createRef} from 'react';
import {
  StyleSheet,
  TextInput,
  View,
  Text,
  Image,
  KeyboardAvoidingView,
  Keyboard,
  ToastAndroid,
  TouchableOpacity,
  AsyncStorage,
  ScrollView,
} from 'react-native';
import {COLORS} from '../theme/colors';
import {AXIOS_CLIENT} from '../services/axiosCientService';
import {TEXT, CONFIG, KEYS} from '../constants';

export default function Login({navigation}) {
  const [userName, setUserName] = useState('');
  const [password, setPassword] = useState('');

  const passwordInputRef = createRef();

  useEffect(() => {
    checkToken();
  }, []);

  const checkToken = async () => {
    let token = await AsyncStorage.getItem(KEYS.TOKEN_KEY);
    if (token) {
      navigateMain();
    }
  };

  const handleLogin = () => {
    Keyboard.dismiss();
    if (CONFIG.skipLogin) {
      AsyncStorage.setItem(KEYS.USER_NAME_KEY, userName);
      navigateMain();
    } else if (userName.length > 0 && password.length > 0) {
      requestLogin();
    } else {
      ToastAndroid.show(TEXT.loginMissingInput, ToastAndroid.SHORT);
    }
  };

  const requestLogin = () => {
    AXIOS_CLIENT.post('login', {
      data: {
        userName: userName,
        password: password,
      },
    })
      .then(response => {
        if (response.status === 200) {
          AsyncStorage.setItem(KEYS.TOKEN_KEY, response.data.token);
          AsyncStorage.setItem(KEYS.USER_NAME_KEY, userName);
          navigateMain();
        } else if (
          response.status === 400 ||
          response.status === 401 ||
          response.status === 405
        ) {
          ToastAndroid.show(response.data.message, ToastAndroid.SHORT);
        } else {
          ToastAndroid.show(TEXT.unexpectedError, ToastAndroid.SHORT);
        }
      })
      .catch(error => {
        console.info(error);
        ToastAndroid.show(TEXT.networkError, ToastAndroid.SHORT);
      });
  };

  const navigateMain = async () => {
    navigation.navigate('Home');
  };

  const navigateRegister = async () => {
    Keyboard.dismiss();
    navigation.navigate('Registration');
  };

  return (
    // eslint-disable-next-line react-native/no-inline-styles
    <View style={{flex: 1}}>
      <ScrollView
        keyboardShouldPersistTaps="handled"
        styles={styles.contentContainerStyle}>
        <View style={{alignItems: 'center'}}>
          <Image
            source={{
              uri: 'https://www.astajans.com/Upload/urunler-upload/img_96.png',
            }}
            style={styles.imageStyle}
          />
        </View>
        <KeyboardAvoidingView enabled>
          <View style={styles.SectionStyle}>
            <TextInput
              style={styles.inputStyle}
              onChangeText={UserName => setUserName(UserName)}
              underlineColorAndroid="#f000"
              placeholder="Email / Kullanıcı Adı"
              placeholderTextColor="#8b9cb5"
              autoCapitalize="sentences"
              returnKeyType="next"
              onSubmitEditing={() =>
                passwordInputRef.current && passwordInputRef.current.focus()
              }
              blurOnSubmit={false}
            />
          </View>
          <View style={styles.SectionStyle}>
            <TextInput
              style={styles.inputStyle}
              ref={passwordInputRef}
              onChangeText={UserPassword => setPassword(UserPassword)}
              underlineColorAndroid="#f000"
              placeholder="Şifre"
              placeholderTextColor="#8b9cb5"
              returnKeyType="next"
              secureTextEntry={true}
              onSubmitEditing={() => Keyboard.dismiss()}
              blurOnSubmit={false}
            />
          </View>
          <TouchableOpacity
            style={styles.buttonStyle}
            activeOpacity={0.5}
            onPress={handleLogin}>
            <View>
              <Text style={styles.buttonTextStyle}>Giriş Yap</Text>
            </View>
          </TouchableOpacity>
          <TouchableOpacity
            style={styles.buttonStyle}
            activeOpacity={0.5}
            onPress={navigateRegister}>
            <View>
              <Text style={styles.buttonTextStyle}>Kayıt Ol</Text>
            </View>
          </TouchableOpacity>
        </KeyboardAvoidingView>
      </ScrollView>
    </View>
  );
}

const styles = StyleSheet.create({
  SectionStyle: {
    flexDirection: 'row',
    height: 40,
    marginTop: 20,
    marginLeft: 35,
    marginRight: 35,
    margin: 10,
  },
  buttonStyle: {
    backgroundColor: COLORS.buttonColor,
    borderWidth: 0,
    color: COLORS.buttonColor,
    borderColor: COLORS.buttonColor,
    height: 40,
    alignItems: 'center',
    borderRadius: 30,
    marginLeft: 35,
    marginRight: 35,
    marginTop: 20,
    marginBottom: 20,
  },
  buttonTextStyle: {
    color: COLORS.buttonTextColor,
    paddingVertical: 10,
    fontSize: 16,
  },
  inputStyle: {
    flex: 1,
    color: COLORS.inputTextColor,
    paddingLeft: 15,
    paddingRight: 15,
    borderWidth: 1,
    borderRadius: 30,
    borderColor: '#dadae8',
  },
  contentContainerStyle: {
    justifyContent: 'center',
    alignContent: 'center',
  },
  imageStyle: {
    width: '50%',
    height: 100,
    resizeMode: 'contain',
    margin: 30,
  },
});
