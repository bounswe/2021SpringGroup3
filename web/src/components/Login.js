import PropTypes from 'prop-types';
import { useState } from 'react';
import { Form, Input, Row, Col, Button, Typography, Layout, Space } from 'antd';

import React from "react";

import { LoginOutlined } from '@ant-design/icons';
import 'antd/dist/antd.css';

const { Text } = Typography;

const subButtonStyle = {
  fontSize: '12px',
  color: '#8b949e',
  cursor: 'pointer'
}

const Login = () => {

  const onFinish = (values) => {
    console.log('Success:', values);
  };

  const onFinishFailed = (errorInfo) => {
    console.log('Failed:', errorInfo);
  };

  const sendToRegistration = () => {
    console.log('Sending to registration');
  }

  const sendToResetPassword = () => {
    console.log('Sending to reset password');
  }

  return (
    <Row gutter={[0, 16]}>
      <Col span={24}></Col>
      <Col span={6} offset={9}>
        <Form
          style={{marginBottom: '20px'}}
          name="basic"
          onFinish={onFinish}
          onFinishFailed={onFinishFailed}
        >
          <Text strong>Username or Email</Text>
          <Form.Item
            name="usernameOrEmail"
            rules={[
              {
                required: true,
                message: 'Please enter your username or email.',
              },
            ]}
          >
            <Input placeholder="Enter username or email" />
          </Form.Item>

          <Text strong>Password</Text>
          <Form.Item
            name="password"
            rules={[
              {
                required: true,
                message: 'Please enter your password.',
              },
            ]}
          >
          <Input.Password placeholder="Enter password" />
          </Form.Item>

          <Col span={24} align="middle">
            <Button type="primary" htmlType="submit" shape="round" icon={<LoginOutlined />}>
                Login
            </Button>
          </Col>
        </Form>
        <Col span={24} align="middle">
          <Space size={'large'}>
            <Text style={subButtonStyle} onClick={sendToResetPassword}>Forgot your password?</Text>
            <Text style={subButtonStyle} onClick={sendToRegistration}>Don't have an account?</Text>
          </Space>
        </Col>
      </Col >
    </Row >
  )
}

export default Login;