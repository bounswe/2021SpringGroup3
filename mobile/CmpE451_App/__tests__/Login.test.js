import React from 'react';
import Login from '../src/screen/Login';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(<Login />).toJSON();
  expect(tree).toMatchSnapshot();
});
