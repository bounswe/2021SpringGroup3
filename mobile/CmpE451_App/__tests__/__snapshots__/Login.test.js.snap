// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    keyboardShouldPersistTaps="handled"
    style={
      Object {
        "alignContent": "center",
        "flex": 6,
        "justifyContent": "center",
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/image/logo.png",
            }
          }
          style={
            Object {
              "height": 100,
              "margin": 30,
              "resizeMode": "contain",
              "width": "50%",
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "height": 40,
            "margin": 10,
            "marginLeft": 35,
            "marginRight": 35,
            "marginTop": 20,
          }
        }
      >
        <TextInput
          autoCapitalize="sentences"
          blurOnSubmit={false}
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="Email / Kullanıcı Adı"
          placeholderTextColor="#8b9cb5"
          returnKeyType="next"
          style={
            Object {
              "borderColor": "#dadae8",
              "borderRadius": 30,
              "borderWidth": 1,
              "color": "#141414",
              "flex": 1,
              "paddingLeft": 15,
              "paddingRight": 15,
            }
          }
          underlineColorAndroid="#f000"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "height": 40,
            "margin": 10,
            "marginLeft": 35,
            "marginRight": 35,
            "marginTop": 20,
          }
        }
      >
        <TextInput
          blurOnSubmit={false}
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="Şifre"
          placeholderTextColor="#8b9cb5"
          returnKeyType="next"
          secureTextEntry={true}
          style={
            Object {
              "borderColor": "#dadae8",
              "borderRadius": 30,
              "borderWidth": 1,
              "color": "#141414",
              "flex": 1,
              "paddingLeft": 15,
              "paddingRight": 15,
            }
          }
          underlineColorAndroid="#f000"
        />
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#3849aa",
            "borderColor": "#3849aa",
            "borderRadius": 30,
            "borderWidth": 0,
            "color": "#3849aa",
            "height": 40,
            "marginBottom": 20,
            "marginLeft": 35,
            "marginRight": 35,
            "marginTop": 20,
            "opacity": 1,
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "color": "#ffffff",
                "fontSize": 16,
                "paddingVertical": 10,
              }
            }
          >
            Login
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#3849aa",
            "borderColor": "#3849aa",
            "borderRadius": 30,
            "borderWidth": 0,
            "color": "#3849aa",
            "height": 40,
            "marginBottom": 20,
            "marginLeft": 35,
            "marginRight": 35,
            "marginTop": 20,
            "opacity": 1,
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "color": "#ffffff",
                "fontSize": 16,
                "paddingVertical": 10,
              }
            }
          >
            Register
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  />
</View>
`;
